distributed/worker.py

```diff
@@ -3697,6 +3703,15 @@ def _notify_plugins(self, method_name, *args, **kwargs):
                         "Plugin '%s' failed with exception", name, exc_info=True
                     )
 
+    async def benchmark_disk(self, comm=None):
+        return self.executor.submit(benchmark_disk).result()
+
+    async def benchmark_memory(self, comm=None):
+        return self.executor.submit(benchmark_memory).result()
+
+    async def benchmark_network(self, comm=None, address=None):
+        return await benchmark_network(rpc=self.rpc, address=address)
```

Comment:
1. please add type annotations
2. comm=None is not necessary anymore, please remove
3. please don't set a default to None for mandatory parameter ``address``
4. benchmark_disk and benchmark_memory block the event loop
5. self.executor is *typically* a ThreadPoolExecutor, unless the user overrides it from the Worker kwargs. I think in this case it should be explicitly a threadpool:

```
def benchmark_disk(self) -> asyncio.Future[...]:
    return self.loop.run_in_executor(None, benchmark_disk)
```
6. this looks like a good opportunity to move these 3 functions to a separate module

Reply:
1. I've added annotations.  However I'm curious, is this something that we're requiring for PRs now or is this just your personal preference?
2. Cool.  I'm glad to see that change.  Removed
3. What do you think the parameter should be?  `""`?
4. I'm confused by this.  We're running them in a separate thread.  Is there something that I'm missing here?
5. This seems overly cautious to me, but I'm happy to make the change 
6.  I thought about that but couldn't find a good place to put them.  Do you have a suggestion?

Reply:
3. it should be mandatory, e.g. without a default.
4. ``return self.executor.submit(benchmark_memory).result()`` is a synchronous instruction that blocks the event loop of the main thread until the future is done.
6. ``distributed.diagnostics.worker_benchmark``?